### ISSUE TITLE
boost160 segmentation fault during build

### ISSUE THREAD

**Issue by gefla** [NixOS/nixpkgs#203080]
### Steps To Reproduce
`curl https://hydra.nixos.org/build/199563599/reproduce | bash`

### Build log
```
building
Performing configuration checks

    - 32-bit                   : no
    - 64-bit                   : yes
    - arm                      : yes

Building the Boost C++ Libraries.


    - symlinks supported       : yes
    - lockfree boost::atomic_flag : yes
    - has_icu builds           : yes
warning: Graph library does not contain MPI-based parallel components.
note: to enable them, add "using mpi ;" to your user-config.jam
    - zlib                     : yes
    - iconv (libc)             : yes
    - icu                      : yes
    - compiler-supports-visibility : yes
    - gcc visibility           : yes
    - long double support      : yes
warning: skipping optional Message Passing Interface (MPI) library.
note: to enable MPI support, add "using mpi ;" to user-config.jam.
note: to suppress this message, pass "--without-mpi" to bjam.
note: otherwise, you can safely ignore this message.
/nix/store/4w2rv6s96fwsb4qyw8b9w394010gxriz-stdenv-darwin/setup: line 1407: 24799 Segmentation fault: 11  b2 --includedir=$dev/include --libdir=$out/lib -j$(($NIX_BUILD_CORES<=256 ? $NIX_BUILD_CORES : 256)) --layout=system variant=release threading=multi link=shared -sEXPAT_INCLUDE=/nix/store/k9gdd15bc7v13h0jza8pi8rkz3r2k0y3-expat-2.5.0-dev/include -sEXPAT_LIBPATH=/nix/store/ysh8hcdnnbm5h90g1pdfllbz36nhc3zj-expat-2.5.0/lib runtime-link=shared debug-symbols=off toolset=clang --without-python --user-config=user-config.jam
 ```

### Additional context
Fails on aarch64

### Notify maintainers
@hjones2199 

### Metadata

```console
[nix-shell:~/src/nixpkgs/boost_1_60_0]$ nix-shell -p nix-info --run "nix-info -m"
these 2 paths will be fetched (0.01 MiB download, 0.14 MiB unpacked):
  /nix/store/0gfg84zy171sgpwfbgwji3ybbjh38rh4-nix-info
  /nix/store/yarklbql3fs5b3r38h3w25nax2s8rnd3-DarwinTools-1
copying path '/nix/store/yarklbql3fs5b3r38h3w25nax2s8rnd3-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/0gfg84zy171sgpwfbgwji3ybbjh38rh4-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.6.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.7.0`
 - channels(gefla): `"home-manager, nixpkgs, unstable"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/Users/gefla/.nix-defexpr/channels/nixpkgs````

**Comment by wegank** [comment]
Is there a use case for boost160? IMHO this will not be fixed as boost 1.60 was released 5 years before M1 existed.

**Comment by gefla** [comment]
It's currently a dependency of mongodb-3_4 and mongodb-3_6. I tried just change that dependency to boost-16x (which currently maps to 1.69), however that breaks the mongodb build.

Nothing in nixpkgs seems to depend on those mongodb versions, so maybe remove those as well?
